packages/shopware-6-client/src/services/customerService.ts

```diff
@@ -144,7 +144,7 @@ export async function getCustomerOrderDetails(
 
   const resp = await contextInstance.invoke.get(getCustomerOrderEndpoint(), {
     // TODO: move into plain Object after https://github.com/DivanteLtd/shopware-pwa/issues/911 is merged
-    params: `filter[id]=${orderId}&associations[lineItems][]&associations[addresses][]&associations[transactions][]&associations[deliveries][]`,
+    params: `filter[id]=${orderId}&associations[lineItems][]&associations[addresses][]&associations[transactions][associations][paymentMethod][]&associations[deliveries][associations][shippingMethod][]`,
```

Comment:
Ordering transaction association by `createdAt DESC` and set `limit: 1` will give you the correct transaction.

Reply:
done :slightly_smiling_face: